content/blogposts/2022/sourcegraph-4.0.md

```diff
@@ -0,0 +1,256 @@
+tags: [blog, release]
+slug: "release/4.0"
+published: true
+heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/4.0/release.png
+socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/4.0/release.png
+changelogItems:
+---
+
+We've spent 9 years building the underlying foundation of code search to be fast, secure, and scalable across ~10^11 lines of code for the world's most demanding software teams. Now, we’re making Sourcegraph even smarter.
+
+Sourcegraph 4.0 includes over a dozen updates and new features, and it is the first step toward Sourcegraph becoming a [code intelligence platform](/blog/code-search-to-code-intelligence). A code intelligence platform indexes and understands all of your organization’s code, so instead of just *matching lines in code*, you can *answer questions and learn about code*. 
+
+Today, we're also launching [Sourcegraph Cloud:](/blog/enterprise-cloud) secure, scalable, dedicated Sourcegraph instances on the cloud. Sourcegraph Cloud is the best way to start using Sourcegraph on your organization's code. [Sign up now](https://signup.sourcegraph.com/) to experience Sourcegraph 4.0.
+
+
+
+**Read on to discover everything in Sourcegraph 4.0:**
+* [🧠 Code intelligence: uplevel your code search](#code-intelligence-uplevel-your-code-search)
+* [🏗️ High-leverage ways to improve your entire codebase](#high-leverage-ways-to-improve-your-entire-codebase)
+* [☁️ Dedicated Sourcegraph Cloud instances for enterprise](#dedicated-sourcegraph-cloud-instances-for-enterprise)
+* [📈 Advanced admin capabilities](#advanced-admin-capabilities)
+
+
+### <a id="code-intelligence-uplevel-your-code-search"></a> 🧠 Code intelligence: uplevel your code search
+
+<br />
+<Badge text="Code Insights" link="/code-insights" color="green" size="small" />
+
+#### Understand usage and code structure with Code Insights in the search UI
+[Code Insights](/code-insights) turns your code into a queryable database to create customizable, visual dashboards. We're bringing the power of Code Insights directly into the search UI to surface high-level information from your codebase, help you answer questions about your code, and refine your search results.  
+
+Search aggregation charts appear on the search results page to answer questions like:
+- How many different versions of a library or package are present in your code, and which is most common? 
+- Which files have the majority of these search results?
+- Which repositories is this library used in most?
+- Which library components are used most? By who?
+- What are all the possible arguments we pass to this function? 
+
+You can choose to group your search results by location (repo or file), author, or arbitrary capture group pattern. Here are some [example search aggregations to get started with.](https://docs.sourcegraph.com/code_insights/references/search_aggregations_use_cases)
```

Comment:
[Content guidelines: use "repository" instead of "repo"](https://handbook.sourcegraph.com/company-info-and-process/communication/content_guidelines/terminology_guidelines/#:~:text=a%20good%20look.-,repository,-repo)

```suggestion
You can choose to group your search results by location (repository or file), author, or arbitrary capture group pattern. Here are some [example search aggregations to get started with.](https://docs.sourcegraph.com/code_insights/references/search_aggregations_use_cases)
```